### PR TITLE
[#1401] feat(dashboard): Shuffle server page list sorting function.

### PR DESCRIPTION
**Pull request by yl09099**
### What changes were proposed in this pull request?

On the Shuffle Server list page, you can sort the Shuffle Server by column to facilitate you to view information about the Shuffle Server.You can sort IP, used memory, pre-allocated memory, available memory, heartbeat time, and so on.
image

### Why are the changes needed?

Fix: #1401 

### Does this PR introduce _any_ user-facing change?

Yes.

### How was this patch tested?

Page view.

### CODE REVIEW

**Comment by yl09099** [comment]
@xianjingfeng @jerqi @rickyma PTAL

**Comment by rickyma** [comment]
Great feature! I hope some of the features in https://github.com/apache/incubator-uniffle/issues/1401 are already on the roadmap. 

cc @maobaolong

**Comment by rickyma** [comment]
This should be a feature instead of an improvement.

File: dashboard/src/main/webapp/src/pages/ApplicationPage.vue [hunk 1]
```diff
@@ -32,19 +32,43 @@
...
+        @sort-change="sortAppChangeEvent"
+      >
+        <el-table-column prop="appId" label="AppId" min-width="180" :sortable="true" />
+        <el-table-column prop="userName" label="UserName" min-width="180" :sortable="true" />
+        <el-table-column
+          prop="registrationTime"
+          label="Registration Time"
+          min-width="180"
+          :formatter="dateFormatter"
+          :sortable="true"
```
Comment (xianjingfeng) [inline dashboard/src/main/webapp/src/pages/ApplicationPage.vue:63]:
```suggestion
          sortable
```

File: dashboard/src/main/webapp/src/pages/serverstatus/ActiveNodeListPage.vue [hunk 2]
```diff
@@ -17,31 +17,46 @@
...
+        prop="usedMemory"
+        label="UsedMem"
+        min-width="80"
+        :formatter="memFormatter"
+        :sortable="true"
+      />
       <el-table-column
         prop="preAllocatedMemory"
-        label="PreAllocatedMem"
+        label="Pre-Mem"
```
Comment (xianjingfeng) [inline dashboard/src/main/webapp/src/pages/serverstatus/ActiveNodeListPage.vue:40]: I think the original name is better.

**Comment by xianjingfeng** [comment]
I think most of codes in the node list pages are the same. I think we should create a component for node list.  WDYT? @yl09099

File: dashboard/src/main/webapp/src/pages/serverstatus/ActiveNodeListPage.vue (hunk 2 above)
Comment (yl09099) [reply dashboard/src/main/webapp/src/pages/serverstatus/ActiveNodeListPage.vue:40]: > I think the original name is better.

It turned out that the name was too long, I had to change lines when the page was displayed, and I looked very uncomfortable.Do you have any good ideas?

**Comment by yl09099** [comment]
> I think most of codes in the node list pages are the same. I think we should create a component for node list. WDYT? @yl09099

That's the right thing to do, but my research.

**Comment by yl09099** [comment]
> I think most of codes in the node list pages are the same. I think we should create a component for node list. WDYT? @yl09099

It's the right thing to do, but I have to study it.

**Comment by yl09099** [comment]
> I think most of codes in the node list pages are the same. I think we should create a component for node list. WDYT? @yl09099

This one's ready. I hope I'm taking a look at it.

File: dashboard/src/main/webapp/src/pages/serverstatus/ActiveNodeListPage.vue (hunk 2 above)
Comment (xianjingfeng) [reply dashboard/src/main/webapp/src/pages/serverstatus/ActiveNodeListPage.vue]: Maybe we can fix the first column.
https://element.eleme.io/#/en-US/component/table#table-with-fixed-column

Comment (yl09099) [reply dashboard/src/main/webapp/src/pages/serverstatus/ActiveNodeListPage.vue]: done

File: dashboard/src/main/webapp/src/pages/serverstatus/ListPage.vue [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+          getShuffleActiveNodesPage()
+        } else if (router.currentRoute.value.name === 'decommissioningNodeList') {
+          getShuffleDecommissioningListPage()
+        } else if (router.currentRoute.value.name === 'decommissionedNodeList') {
+          getShuffleDecommissionedListPage()
+        } else if (router.currentRoute.value.name === 'unhealthyNodeList') {
+          getShuffleUnhealthyListPage()
+        } else if (router.currentRoute.value.name === 'lostNodeList') {
+          getShuffleLostListPage()
+        }
```
Comment (xianjingfeng) [inline dashboard/src/main/webapp/src/pages/serverstatus/ListPage.vue:162]: Could you create a function for this part of code?

File: dashboard/src/main/webapp/src/pages/serverstatus/ListPage.vue
Comment (xianjingfeng) [inline dashboard/src/main/webapp/src/pages/serverstatus/ListPage.vue:1]: `NodeListPage.vue` may be better.

**Review by xianjingfeng — APPROVED** [review]
LGTM

**Comment by rickyma** [comment]
Merged to master.
Thanks for your contribution @yl09099 
Thanks for your review @xianjingfeng

**Pull request merged**